account_invoice_accrual/models/account_invoice.py

```diff
@@ -0,0 +1,297 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, exceptions, fields, models, _
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.model
+    def move_line_get_item(self, line):
+        res = super(AccountInvoiceLine, self)\
+            .move_line_get_item(line,)
+        if self.env.context.get('move_accrual') and self.env.context.get(
+                'type'):
+            mapping_obj = self.env['account.cutoff.mapping']
+            if self.env.context.get('type', False) in ['out_invoice',
+                                                       'out_refund']:
+                mapping_type = 'accrued_revenue'
+            else:
+                mapping_type = 'accrued_expense'
+            mapping = mapping_obj.\
+                _get_mapping_dict(line.invoice_id.company_id.id,
+                                  mapping_type)
+            res['account_id'] = mapping.get(line.account_id.id,
+                                            line.account_id.id)
+        return res
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    accrual_move_id = fields.Many2one(
+        'account.move', 'Accrual Journal Entry',
+        readonly=True, ondelete='set null', copy=False,
+        help="Link to the Accrual Journal Items.")
+    to_be_reversed = fields.Boolean(
+        'To be reversed', related='accrual_move_id.to_be_reversed')
+
+    def reverse_accruals(self):
+        for invoice in self:
+            if not invoice.to_be_reversed:
+                continue
+            accrual_move = invoice.accrual_move_id
+            accrual_date = fields.Date.from_string(accrual_move.date)
+            date = self.date or self.date_invoice
+            invoice_date = fields.Date.from_string(date)
+
+            if (accrual_move.state == 'draft' and
+                    accrual_date.year == invoice_date.year and
+                    accrual_date.month == invoice_date.month):
+                # reversal in same month as accrual
+                # and accrual move is still draft:
+                # we simply remove it
+                accrual_move.unlink()
+            else:
+                # Use default values of the reversal wizard to create the
+                # reverse
+                reverse_obj = self.env['account.move.reverse']
+                reverse_wizard =\
+                    reverse_obj.create({'date': invoice_date})
```

Comment:
Can you also add 'journal_id': accrual_move.journal_id.id
Otherwise, the accrual reversal is in the journal of the invoice and not in the journal of the accrual.